microservices/coco-attestation/attestation-module-snpguest/src/main/java/com/suse/coco/module/snpguest/SNPGuestWorker.java

```diff
@@ -0,0 +1,125 @@
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+
+package com.suse.coco.module.snpguest;
+
+import com.suse.coco.model.AttestationResult;
+import com.suse.coco.module.snpguest.execution.SNPGuestWrapper;
+import com.suse.coco.module.snpguest.io.VerificationDirectoryProvider;
+import com.suse.coco.module.snpguest.model.AttestationReport;
+import com.suse.coco.module.snpguest.model.EpycGeneration;
+import com.suse.coco.modules.AttestationWorker;
+import com.suse.common.io.ByteSequenceFinder;
+
+import org.apache.ibatis.session.SqlSession;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.nio.file.Path;
+
+/**
+ * Worker class for verifying the reports with SNPGuest
+ */
+public class SNPGuestWorker implements AttestationWorker {
+
+    private static final Logger LOGGER = LogManager.getLogger(SNPGuestWorker.class);
+
+    private final VerificationDirectoryProvider directoryProvider;
+
+    private final SNPGuestWrapper snpGuest;
+
+    private final ByteSequenceFinder sequenceFinder;
+
+    /**
+     * Default constructor.
+     */
+    public SNPGuestWorker() {
+        this(new VerificationDirectoryProvider(), new SNPGuestWrapper(), new ByteSequenceFinder());
+    }
+
+    /**
+     * Constructor with explicit dependencies, for unit test only.
+     * @param directoryProviderIn the verification directory provider
+     * @param snpGuestWrapperIn the snpguest executor
+     * @param sequenceFinderIn the byte sequence finder
+     */
+    SNPGuestWorker(VerificationDirectoryProvider directoryProviderIn, SNPGuestWrapper snpGuestWrapperIn,
+                   ByteSequenceFinder sequenceFinderIn) {
+        this.directoryProvider = directoryProviderIn;
+        this.snpGuest = snpGuestWrapperIn;
+        this.sequenceFinder = sequenceFinderIn;
+    }
+
+    @Override
+    public boolean process(SqlSession session, AttestationResult result) {
+
+        try {
+            LOGGER.debug("Processing attestation result {}", result.getId());
+
+            AttestationReport report = session.selectOne("SNPGuestModule.retrieveReport", result.getReportId());
+            if (report == null) {
+                LOGGER.error("Unable to retrieve attestation report for result {}", result.getId());
+                return false;
+            }
+
+             LOGGER.debug("Loaded report {}", report);
+             if (report.getCpuGeneration() == EpycGeneration.UNKNOWN) {
+                LOGGER.error("Unable to identify Epyc processor generation for attestation report {}", report.getId());
+                return false;
+            }
+
+            try (var workingDir = directoryProvider.createDirectoryFor(result.getId(), report)) {
```

Comment:
Does this directory need to be clean/removed after processing?

Reply:
The cleanup is currently done [in the `close()` method of `VerificationDirectory`](https://github.com/uyuni-project/uyuni/blob/7c04887039ba03ca2cfa110ae416068786dd083c/microservices/coco-attestation/attestation-module-snpguest/src/main/java/com/suse/coco/module/snpguest/io/VerificationDirectory.java#L71-L92) and called outmatically by the `try-with-resource`.